### DISCUSSION TITLE
In the traditional one­-way nesting, an error occurred ww3_shel.

### DISCUSSION THREAD

**Discussion by nYifeiLee** [NOAA-EMC/WW3#1222] [Q&A]
Hi everyone！
My large scale grid has a resolution of 10′ and a small scale grid has a resolution of 6 ′.The computer I'm using is a virtual machine running 34.8GB of RAM.When I run the ww3_shel on a small scale grid, WW3 gets an error.In another previous simulation, WW3 also prompted this error due to my too fine meshing.I don't know if it's because my computer is running too little memory or something else that caused this error.
Below is a picture of the WW3 error prompting, along with my input file.I hope you can give me some advice, thank you!

ERROR:
![Ubuntu 22 - VMware Workstation 2024_4_25 21_48_01](https://github.com/NOAA-EMC/WW3/assets/167869457/7bd61655-677b-48bc-ad56-e6a868ae6bb2)

Large scale grid:
[ww3_grid.inp.txt](https://github.com/NOAA-EMC/WW3/files/15111440/ww3_grid.inp.txt)
[ww3_prnc.inp.txt](https://github.com/NOAA-EMC/WW3/files/15111452/ww3_prnc.inp.txt)
[ww3_shel.inp.txt](https://github.com/NOAA-EMC/WW3/files/15111486/ww3_shel.inp.txt)
Small scale grid：
[ww3_grid.inp.txt](https://github.com/NOAA-EMC/WW3/files/15111523/ww3_grid.inp.txt)
[ww3_prnc.inp.txt](https://github.com/NOAA-EMC/WW3/files/15111537/ww3_prnc.inp.txt)
[ww3_shel.inp.txt](https://github.com/NOAA-EMC/WW3/files/15111562/ww3_shel.inp.txt)

**Comment by ukmo-ccbunney** [comment]
Hi @nYifeiLee 
Is there any other output to the screen?
Sometimes the actual wave model error message can be hidden amongst a lot of those w3srervmd message lines...
MPI_ABORT is being called with error code 12, which could help to narrow down the internal error (there are a few modules that use that error code).

**Reply by ukmo-ccbunney** [reply]
One of the modules that throws and error code 12 is the boundary i/o module (w3iobcmd). It could be that one of your boundary points is not located correctly?

**Reply by nYifeiLee** [reply]
Dear Chris Bunney
Thanks for the suggestion, I'll go check the settings of the boundary points again. I will communicate with you in time for the follow-up!

**Reply by ukmo-ccbunney** [reply]
Is there any more output to the screen? The error message might be hidden there...

**Reply by nYifeiLee** [reply]
This is the output of the screen after performing the ww3_shel.
![Ubuntu 22 - VMware Workstation 2024_4_26 17_34_56](https://github.com/NOAA-EMC/WW3/assets/167869457/12fc4840-ad02-406e-87c0-406619f60095)

**Reply by nYifeiLee** [reply]
When I perform a ww3_gird of a small scale grid, the screen shows "2" as the valid boundary point.Does this mean that there is no problem with the boundary points of the input?This is shown in the figure below.
![Ubuntu 22 - VMware Workstation 2024_4_26 17_32_53](https://github.com/NOAA-EMC/WW3/assets/167869457/7f6c5e58-ac00-4f49-8885-5d25192b8262)
![Ubuntu 22 - VMware Workstation 2024_4_26 17_33_24](https://github.com/NOAA-EMC/WW3/assets/167869457/6155a161-b32f-4807-af6f-61d185cd15e5)
![Ubuntu 22 - VMware Workstation 2024_4_26 17_33_40](https://github.com/NOAA-EMC/WW3/assets/167869457/da6dd163-0dbb-4803-8c78-e3a2b1e665e3)
![Ubuntu 22 - VMware Workstation 2024_4_26 17_34_00](https://github.com/NOAA-EMC/WW3/assets/167869457/941e5258-9c56-4578-afc3-61ae255c6333)

**Reply by nYifeiLee** [reply]
Since it was not possible to select the desired resolution when downloading the bathymetric data on the ETOPO, the 1' bathymetric data was downloaded on the ETOPO and then processed to the desired resolution. There may be a problem with the boundaries of the processing. For example, the right boundary of my large-scale model output is a point at 134°E, while the small-scale model only recognizes 133.9°E on its right boundary when performing ww3_grid due to the change in resolution. So I think that's where the problem arises. Do you think my understanding is correct?

**Reply by ukmo-ccbunney** [reply]
Is you screenshot of the output from ww3_shel the entirety of the output to the screen?

If your output boundary locations from your course model do not align with the input BCs in the nested model (within a very small tolerance), then this could be the issue.

**Reply by nYifeiLee** [reply]
Thanks for the reminder!I set the output boundary and nested boundary to be consistent and then rerun the ww3_shel.See if this issue still arises.
Below is a screenshot of what ww3_shel output is all about.
![Ubuntu 22 - VMware Workstation 2024_4_26 21_34_10](https://github.com/NOAA-EMC/WW3/assets/167869457/d46f518d-2f77-4adf-89a3-c88c1ff5227c)
![Ubuntu 22 - VMware Workstation 2024_4_26 17_34_56](https://github.com/NOAA-EMC/WW3/assets/167869457/657da3c1-7040-484e-8813-090d557b3d33)

**Reply by nYifeiLee** [reply]
Unfortunately, I still get this error after I set the boundaries of the output mesh to be the same as the nested grid, but "IEXIT=12" becomes "IEXIT=20".I now doubt if it's because my computer runs too little memory to run this nested computation smoothly.

**Reply by ukmo-ccbunney** [reply]
Sounds like a different error now. 
Do you have the output of STDERR? Usually the model will output a useful message before exiting with an error code.
You model is not crashing, it is doing a controlled exit, so it should be telling you why.

Normally, you would see a line output like this:
`*** WAVEWATCH III ERROR ... `

**Reply by nYifeiLee** [reply]
It's a pity that I didn't see a "*** WAVEWATCH III ERROR ...  ".This screenshot is the content of "log.ww3".
![Ubuntu 22 - VMware Workstation 2024_4_27 19_34_47](https://github.com/NOAA-EMC/WW3/assets/167869457/674d0baa-2372-4f92-9d2b-5208eb3543d9)

The CPU of my computer is i7-8700 and it is too old.I'm going to run this model with a better computer.

**Reply by aliabdolali** [reply]
Just as a sanity check, run your smaller scale domain w/o boundary/nest file and see if it runs or not. If it runs, the problem is with thr boundary. Usually,you would see problems with boundary if you have open boundary nodes in shallow depth while they are not well captured in the large scale parent simulation. 
The easy fix is to exclude those open boundary nodes from your small scale domainz

**Reply by nYifeiLee** [reply]
Dear aliabdolali
Thanks for the suggestion.I ran a small area with no nested borders and it worked correctly.So, as you predicted, the problem should arise at the boundary point.Could you please elaborate on the meaning of the term "open boundary nodes in shallow depth" so that I can better understand this issue.What does "open boundary nodes in shallow depth" refer to? 
Below is a screenshot of the latest error report.
![353 011 363 2024_5_9 19_22_49](https://github.com/NOAA-EMC/WW3/assets/167869457/ec399779-4487-4476-917d-b5d57c61f3b4)

**Reply by aliabdolali** [reply]
well, I think the depths at open boundary nodes are not ~similar between the parent and child domains, due to the course resolution of the parent domain, so you can either exclude those points as open boundary nodes (which is a safe option, usually nothing gets to your domain from there and if it gets there, it is not representative) or do not use the spec files from those points from the large scale simulation.

**Reply by nYifeiLee** [reply]
I'd like to ask you how nested boundaries should be set up is prescriptive.I read the technical manual before, and my understanding was too superficial.I think that keeping the increments (DX, DY) of the large output boundaries consistent with the small range of resolutions will do the nesting smoothly.But so far, my understanding should be incorrect.I would like you to explain in detail how to set a large range of output boundaries and a small range of input boundaries to ensure that nesting goes smoothly.

**Reply by ZOUABBAS** [reply]
helllo how did you solved the problem, because i have the same
